### ISSUE TITLE
Finch vm init --cpu=4 --memory=4G

### ISSUE THREAD

**Issue by munntjlx** [runfinch/finch#683]
**What is the problem you're trying to solve?.**
No cli options for altering the vm state (aka memory etc) seem to be documented in finch vm init

**Describe the feature you'd like**
If there are flags such as --cpu and --memory this would be extremely useful.

**Additional context**
Customizing the memory and cpu# would be especially useful.

**Comment by austinvazquez** [comment]
Hi @munntjlx thanks for the interest in Finch.

Just as a call out the VM's CPU and memory can be manually changed via Finch's configuration file before the VM initializes. I am using the default Homebrew installation and for me this file is `~/.finch/finch.yaml`.

**Comment by munntjlx** [comment]
I did find that, I just think the cli thing would be VERY handy....highly optional though in light of this discovery!

**Comment by haytok** [comment]
Hi, @austinvazquez 

I am interested in this issue, and working on this feature to execute `finch vm init --cpu ... --memory ...` and `finch vm start --cpu ... --memory ...`. Then, I have a question to resolve this issue of feature request.

**Can we resolve this issue when executing `finch vm init` or `finch vm run` like below results of my development?**

<details><summary>Details</summary>
<p> 

Settings for `finch.yaml`

```bash
haytok finch [main]
> cat ~/.finch/finch.yaml
cpus: 6
memory: 4GiB
vmType: qemu
rosetta: false
```

Details of `finch vm init` with `--cpus` and `--memory` options

```bash
haytok finch [main]
> _output/bin/finch vm init -h
Initialize the virtual machine

Usage:
  finch vm init [flags]

Flags:
      --cpus int        the amount of vCPU to dedicate to the virtual machine (default 2)
  -h, --help            help for init
      --memory string   the amount of memory to dedicate to the virtual machine (default "4GiB")

Global Flags:
      --debug   running under debug mode

```

Execute `finch vm init` with `--cpus` and `--memory` options and Check CPUs and memory allocated for VM

```bash
haytok finch [main]
> _output/bin/finch vm init --cpus 4 --memory 6GiB
INFO[0000] Initializing and starting Finch virtual machine...
INFO[0070] Finch virtual machine started successfully
haytok finch [main]
> LIMA_HOME=/Users/haytok/finch/_output/lima/data/ /Users/haytok/finch/_output/lima/bin/limactl shell finch grep cpu.cores /proc/cpuinfo | sort -u
cpu cores	: 4
haytok finch [main]
> LIMA_HOME=/Users/haytok/finch/_output/lima/data/ /Users/haytok/finch/_output/lima/bin/limactl shell finch free -h
               total        used        free      shared  buff/cache   available
Mem:           5.8Gi       255Mi       4.5Gi        12Mi       1.0Gi       5.3Gi
Swap:          5.8Gi          0B       5.8Gi
```

Details of `finch vm start` with `--cpus` and `--memory` options

```bash
haytok finch [main]
> _output/bin/finch vm start -h
Start the virtual machine

Usage:
  finch vm start [flags]

Flags:
      --cpus int        the amount of vCPU to dedicate to the virtual machine
  -h, --help            help for start
      --memory string   the amount of memory to dedicate to the virtual machine

Global Flags:
      --debug   running under debug mode
```

Execute `finch vm start` with `--cpus` and `--memory` options and Check CPUs and memory allocated for VM

```bash
haytok finch [main]
> _output/bin/finch vm start --cpus 8 --memory 8GiB
INFO[0000] Starting existing Finch virtual machine...
INFO[0055] Finch virtual machine started successfully
haytok finch [main]
> LIMA_HOME=/Users/haytok/finch/_output/lima/data/ /Users/haytok/finch/_output/lima/bin/limactl shell finch grep cpu.cores /proc/cpuinfo | sort -u
cpu cores	: 8
haytok finch [main]
> LIMA_HOME=/Users/haytok/finch/_output/lima/data/ /Users/haytok/finch/_output/lima/bin/limactl shell finch free -h
               total        used        free      shared  buff/cache   available
Mem:           7.7Gi       273Mi       6.9Gi        12Mi       618Mi       7.2Gi
Swap:          7.7Gi          0B       7.7Gi
```

Note that check CPUs and memory allocated for VM when executing `finch vm start` not using `--cpus` and `--memory` options

```bash
haytok finch [main]
> LIMA_HOME=/Users/haytok/finch/_output/lima/data/ /Users/haytok/finch/_output/lima/bin/limactl shell finch grep cpu.cores /proc/cpuinfo | sort -u
cpu cores	: 6
haytok finch [main]
> LIMA_HOME=/Users/haytok/finch/_output/lima/data/ /Users/haytok/finch/_output/lima/bin/limactl shell finch free -h
               total        used        free      shared  buff/cache   available
Mem:           3.8Gi       226Mi       3.0Gi        12Mi       617Mi       3.4Gi
Swap:          3.8Gi          0B       3.8Gi
```

</p>
</details>

**Comment by Shubhranshu153** [comment]
@haytok 
i think i good design will be to have a separate cli option. this gives a cleaner architecture.

```
finch vm settings [args]
```

There are cases where you would need to convey the user, that vm needs to be restarted to take effect and so on.

**Comment by haytok** [comment]
Thanks for comments !!!

I will add subcommand `finch vm setting` to set CPU and memory values.

**Comment by Shubhranshu153** [comment]
Feature has been merged and would be part of the next release 1.1.4
